### DISCUSSION TITLE
How can I get apps to display a system tray icon?

### DISCUSSION THREAD

**Discussion by danirabbit** [elementary/triage#628] [Q&A] [answered]
Many 3rd party applications like Java, NodeWebkit, and Gtk2 applications don't seem to be able to display a system tray icon. Why is that? Is there a workaround?

**Comment by danirabbit** [comment, marked as answer]
**elementary OS does not support GtkStatusIcon (commonly referred to as system tray), or the Ayatana Indicator API.**

elementary OS shows indicator icons on the panel for system status and functions. This area is not intended to be extensible by third-party apps. 

GtkStatusIcon (the old API for apps to place icons in the panel) has been deprecated since 2014 and is not implemented in elementary OS. As of 0.4 Loki, the panel no longer implements the Ayatana API either. It is recommended in the elementary HIG that apps not explicitly use the indicator area at all, as it is reserved for use by the system.

In addition, "system tray" APIs have always been optional and not implemented on all OSes; it was never guaranteed that a FreeDesktop operating system would implement it. As such, we recommend filing a report with application developers still using these APIs requesting that they make their app function without it.

**App developers should instead use modern, cross-desktop APIs for platform integration**

If you're an app developer, you can find lots of information about [well-supported cross-desktop APIs for system integration](https://islinuxabout.xyz/systray/), and read more on the elementary blog post about [Backgrounding & System Integration](https://blog.elementary.io/developer-tips-backgrounding-system-integration/). I also [gave a talk at LAS 2019](https://youtu.be/fPFdV-Z69Lo) about the motivations behind moving away from the system tray concept.

**Reply by davidak** [reply]
FYI: There is a discussion about a [new systemtray spec](https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/84) and GNOME is working on a [Background apps design](https://gitlab.gnome.org/Teams/Design/os-mockups/-/issues/191) ([implementation](https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2624)).

I landed here, because ActivityWatch complains that there is no systray.

**Reply by danirabbit** [reply]
@davidak as of OS 8.1, we implement the XDG background portal, same as GNOME

**Comment by doobes** [comment]
Wow, 

That kinda sucks. 

More regression in v8.0....

**Reply by danirabbit** [reply]
This post is 4 years old. Again, elementary OS never support GtkStatusIcon and Ayatana API was removed in version 0.4 which was released in 2016. There have been no changes whatsoever to this in 8 years

**Comment by doobes** [comment]
I hate to break it to you, but this worked in v7.1....

**Reply by desiderantes** [reply]
Nope this hasn't been available for a while. You might have used some fork or third party plugin offering something like this, but it hasn't been supported for a while.

**Reply by teamcons** [reply]
Moreover, there has been little changes to Wingpanel. The unofficial plugin i think of, that you may have used, should still work the same.

again, please, doobes, a little bit of good faith.

**Comment by enigmaticious** [comment]
@danirabbit I love the argumentation here that "we should urge app developers to change the behaviour of their applications to accommodate the requirements of an OS that 0.1% of the world uses."

I really don't understand the idea strictly imposing of rules of what users can do and cannot do, as much as I don't understand the idea of acting as an universal arbitrator for all users.

I have solved this problem by moving to Zorin OS.

**Comment by SebastianBugReporter** [comment]
I find it a bit unfortunate that suggested changes (https://github.com/elementary/settings-desktop/issues/481#issue-3917400231) that are also useful for some users are being dismissed in this way.
This is similar to my request () asking whether it would be possible to open files with a single click, as is the case with other desktop environments.
